adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenSummaryCheckoutStepController.java

```diff
@@ -209,11 +238,15 @@ public String authorise3DSecurePayment(final Model model,
             return redirectToOrderConfirmationPage(orderData);
         } catch (AdyenNonAuthorizedPaymentException e) {
             PaymentResult paymentResult = e.getPaymentResult();
+            PaymentsResponse paymentsResponse = e.getPaymentsResponse();
             LOGGER.debug("AdyenNonAuthorizedPaymentException with paymentResult: " + paymentResult);
 
-            if (paymentResult.isRefused()) {
+            if (paymentResult != null && paymentResult.isRefused()) {
                 errorMessage = getErrorMessageByRefusalReason(paymentResult.getRefusalReason());
             }
```

Comment:
At a given time, only paymentResult or paymentsResponse will be present hence only one "if" condition should be checked. instead of sequential 'if', if-else can be added.